internal/controller/nonadminbackup_controller.go

```diff
@@ -69,44 +72,98 @@ const (
 func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	r.Log = log.FromContext(ctx)
 	logger := r.Log.WithValues("NonAdminBackup", req.NamespacedName)
+	logger.V(1).Info(">>> Reconcile NonAdminBackup - loop start")
 
+	// Resource version
+	// Generation version - metadata - only one is updated...
 	r.Context = ctx
 	r.NamespacedName = req.NamespacedName
 
+	// Get the Non Admin Backup object
 	nab := nacv1alpha1.NonAdminBackup{}
 	err := r.Get(ctx, req.NamespacedName, &nab)
 
-	if err != nil && errors.IsNotFound(err) {
-		logger.V(1).Info("Deleted NonAdminBackup CR", nameField, req.Name, "Namespace", req.Namespace)
+	// Bail out when the Non Admin Backup reconcile was triggered, when the NAB got deleted
+	// Reconcile loop was triggered when Velero Backup object got updated and NAB isn't there
+	if err != nil && apierrors.IsNotFound(err) {
+		logger.V(1).Info("Deleted NonAdminBackup CR", nameField, req.Name, constant.NameSpaceString, req.Namespace)
```

Comment:
Unsure about this `Deleted` log. This is implying that CR triggred the reconcile but in between reconcile trigger and CR Get call, the CR was deleted. Maybe we just say the CR was not found and not `Deleted`